hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestInsertTable.scala

```diff
@@ -2081,6 +2082,16 @@ class TestInsertTable extends HoodieSparkSqlTestBase {
     })
   }
 
+  // add a listener for stages for parallelism checking with stage name
+  class StageParallelismListener(var stageName: String) extends SparkListener {
+    override def onStageSubmitted(stageSubmitted: SparkListenerStageSubmitted): Unit = {
+      if (stageSubmitted.stageInfo.name.contains(stageName)) {
+        assertResult(1)(stageSubmitted.stageInfo.numTasks)
```

Comment:
Should the count be configurable ?

Reply:
This scenario just need the count equal to 1，otherwise affirm partition pruning is not accurate. @bvaradar